docs/src/api/class-locatorassertions.md

```diff
@@ -757,41 +757,91 @@ expect(locator).to_contain_text(re.compile(r"\d messages"))
 ```
 
 ```csharp
-var locator = Page.Locator(".title");
+var locator = page.Locator(".title");
 await Expect(locator).ToContainTextAsync("substring");
 await Expect(locator).ToContainTextAsync(new Regex("\\d messages"));
 ```
 
-Note that if array is passed as an expected value, entire lists of elements can be asserted:
+If you pass an array as an expected value, the expectations are:
+1. Locator matches a list of elements.
```

Comment:
I had to re-read this a few times. I think it comes down to the fact that:

1. Locators are associated with the idea of "strictness" (that's their main marketing), but these assertions all rely on them resolving to lists. I don't think it's always clear to users that Locators are lazy and it's only when you make an action or assertion involving them that they get resolved and that's when the strictness comes into play.
2. "matches" vs. "resolves": This says _Locator_ **matches** list of _elements_. But, I think _Locator_ **resolves to** a list of _elements_. And _elements_ **match** an array of text.

I think this is all improved by carefully using the words **resolve** and **match**. Reserve **resolve** for Locators since _resolve to_ implies laziness and taking from abstract to concrete elements (plus it mirrors the language in our error messages _locator.click: Error: strict mode violation: "li" resolved to 2 elements:_), and reserve **match** for once you have concrete elements, they can match some text.